src/BaseActionWatcher.ts

```diff
@@ -28,26 +37,74 @@ export class BaseActionWatcher {
    * Start a polling loop
    */
   public async watch(isReplay: boolean = false) {
-    const startTime = new Date().getTime()
+    try {
+      this.running = true
+      const startTime = new Date().getTime()
+
+      await this.checkForBlocks(isReplay)
+      if (this.shouldPause) {
+        this.running = false
+        this.shouldPause = false
+        this.log.info("Demux paused.")
+        return
+      }
+
+      const endTime = new Date().getTime()
+      const duration = endTime - startTime
+      let waitTime = this.pollInterval - duration
+      if (waitTime < 0) {
+        waitTime = 0
+      }
+
+      setTimeout(async () => await this.watch(false), waitTime)
+    } catch (err) {
+      this.running = false
+      this.shouldPause = false
+      this.log.error(err)
+      this.log.info("Demux stopped due to error.")
```

Comment:
should we re-throw the error, or is this the top of the execution stack?

Reply:
I don't want it to stop the process, which throwing would do. This way, the error is logged, but we can still hit the demux endpoint to check status, and possibly even resume indexing (via `/start`) if the problem is fixable via external means.